operator/pkg/ingress/envoy_config.go

```diff
@@ -0,0 +1,371 @@
+		&v2alpha1.CiliumEnvoyConfig{},
+		0,
+		cache.ResourceEventHandlerFuncs{},
+		nil,
+	)
+
+	go manager.informer.Run(wait.NeverStop)
+	if !cache.WaitForCacheSync(wait.NeverStop, manager.informer.HasSynced) {
+		return manager, fmt.Errorf("unable to sync envoy configs")
+	}
+	return manager, nil
+}
+
+// getByKey is a wrapper of Store.GetByKey but with concrete Endpoint object
+func (em *envoyConfigManager) getByKey(key string) (*v2alpha1.CiliumEnvoyConfig, bool, error) {
+	objFromCache, exists, err := em.store.GetByKey(key)
+	if objFromCache == nil || !exists || err != nil {
+		return nil, exists, err
+	}
+	envoyConfig, ok := objFromCache.(*v2alpha1.CiliumEnvoyConfig)
+	if !ok {
+		return nil, exists, fmt.Errorf("got invalid object from cache")
+	}
+	return envoyConfig, exists, err
+}
+
+func getCECNameForIngress(ingress *slim_networkingv1.Ingress) string {
+	return ciliumIngressPrefix + ingress.Namespace + "-" + ingress.Name
+}
+
+func getSecret(k8sClient kubernetes.Interface, namespace, name string) (string, string, error) {
+	secret, err := k8sClient.CoreV1().Secrets(namespace).Get(context.Background(), name, metav1.GetOptions{})
+	if err != nil {
+		return "", "", fmt.Errorf("failied to get secret %s/%s: %v", namespace, name, err)
+	}
+	var tlsKey, tlsCrt []byte
+	var ok bool
+	if tlsKey, ok = secret.Data["tls.key"]; !ok {
+		return "", "", fmt.Errorf("missing tls.key field in secret: %s/%s", namespace, name)
+	}
+	if tlsCrt, ok = secret.Data["tls.crt"]; !ok {
+		return "", "", fmt.Errorf("missing tls.crt field in secret: %s/%s", namespace, name)
+	}
+	return string(tlsCrt), string(tlsKey), nil
+}
+
+func getTLS(k8sClient kubernetes.Interface, ingress *slim_networkingv1.Ingress) (map[string]*envoy_config_core_v3.TransportSocket, error) {
+	tls := make(map[string]*envoy_config_core_v3.TransportSocket)
+	for _, tlsConfig := range ingress.Spec.TLS {
+		crt, key, err := getSecret(k8sClient, ingress.Namespace, tlsConfig.SecretName)
+		if err != nil {
+			return nil, err
+		}
+		for _, host := range tlsConfig.Hosts {
+			downStreamContext := envoy_extensions_transport_sockets_tls_v3.DownstreamTlsContext{
+				CommonTlsContext: &envoy_extensions_transport_sockets_tls_v3.CommonTlsContext{
+					TlsCertificates: []*envoy_extensions_transport_sockets_tls_v3.TlsCertificate{
+						{
+							CertificateChain: &envoy_config_core_v3.DataSource{
+								Specifier: &envoy_config_core_v3.DataSource_InlineString{
+									InlineString: crt,
+								},
+							},
+							PrivateKey: &envoy_config_core_v3.DataSource{
+								Specifier: &envoy_config_core_v3.DataSource_InlineString{
+									InlineString: key,
+								},
+							},
+						},
+					},
+				},
+			}
+			upstreamContextBytes, err := proto.Marshal(&downStreamContext)
+			if err != nil {
+				return nil, err
+			}
+			tls[host] = &envoy_config_core_v3.TransportSocket{
+				Name: "envoy.transport_sockets.tls",
```

Comment:
@sayboras I see you found out that this needs to be "envoy.transport_sockets.tls" instead of just "tls". The bare "tls" name has been deprecated for a long time not, but I was not able to pinpoint the change in Envoy that would have removed the support for this deprecated name.

Reply:
I noticed this during testing only, the current envoy proxy is no longer support 'tls' name only, as sooner or later we need to update this name, so I _didn't_ look deeper to pinpoint which commit/version removed it.